agent/engine/docker_task_engine.go

```diff
@@ -687,14 +687,21 @@ func (engine *DockerTaskEngine) cleanupPauseContainerNetwork(task *api.Task, con
 		return errors.Wrapf(err, "engine: failed cleanup task network namespace, task: %s", task.String())
 	}
 
-	return engine.cniClient.CleanupNS(cniConfig)
+	return engine.cniClient.CleanupNS(cniConfig, engine.cfg.AWSVPCAdditionalLocalRoutes)
 }
 
 func (engine *DockerTaskEngine) buildCNIConfigFromTaskContainer(task *api.Task, container *api.Container) (*ecscni.Config, error) {
 	cfg, err := task.BuildCNIConfig()
 	if err != nil {
 		return nil, errors.Wrapf(err, "engine: build cni configuration from taskfailed")
 	}
+
+	if engine.cfg.OverrideAWSVPCLocalIPv4Address != nil &&
```

Comment:
[nit]
consider moving this expression to a smaller function. ie, `hasValidAWSVPCLocalOverrides(cfg *Config)`

Reply:
Okay.

Reply:
I think I'm going to leave this as-is for now unless you feel strongly about it.